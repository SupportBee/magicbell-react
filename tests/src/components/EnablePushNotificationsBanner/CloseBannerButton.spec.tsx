import { render, RenderResult } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import CloseBannerButton from '../../../../src/components/EnablePushNotificationsBanner/CloseBannerButton';

describe('components', () => {
  describe('CloseBannerButton', () => {
    let onClick: jest.Mock;
    let view: RenderResult;

    beforeEach(() => {
      onClick = jest.fn();

      view = render(<CloseBannerButton onClick={onClick} />);
    });

    afterEach(() => {
      view.unmount();
    });

    describe('render', () => {
      it('renders a button with a close icon', () => {
        expect(view.container).toMatchSnapshot();
      });
    });

    describe('.handleClick', () => {
      it('calls the onClick callback', () => {
        const button = view.getByRole('button');
        userEvent.click(button);

        expect(onClick).toHaveBeenCalledTimes(1);
      });
    });
  });
});
