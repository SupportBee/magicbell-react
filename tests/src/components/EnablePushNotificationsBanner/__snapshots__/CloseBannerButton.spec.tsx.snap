// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components CloseBannerButton render renders a button with a close icon 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 svg {
  height: 24px;
}

<div>
  <button
    class="emotion-0"
  >
    <svg
      fill="none"
      stroke="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 18L18 6M6 6l12 12"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </button>
</div>
`;
